### ISSUE TITLE
Datagrid gets transposed (axis are swapped) when plugin is changed from "Grid" to "Heatmap"

### ISSUE THREAD

**Issue by pgagarinov** [finos/perspective#1127]
#### 1. 
```
import pandas as pd
import numpy as np
from perspective import PerspectiveWidget
from datetime import timedelta,date
arrays = {'A':['bar', 'bar', 'bar', 'bar', 'baz', 'baz', 'baz', 'baz', 'foo', 'foo', 'foo', 'foo', 'qux', 'qux', 'qux', 'qux'],
          'B':['one', 'one', 'two', 'two', 'one', 'one', 'two', 'two', 'one', 'one', 'two', 'two', 'one', 'one', 'two', 'two'],
          'C':['X', 'Y', 'X', 'Y', 'X', 'Y', 'X', 'Y', 'X', 'Y', 'X', 'Y', 'X', 'Y', 'X', 'Y'],
          'D':np.arange(date(2020,6,1),date(2020,6,17),timedelta(days=1),dtype=date)}

df = pd.DataFrame(arrays)
df.set_index(['D'])
```
<img width="165" alt="image" src="https://user-images.githubusercontent.com/4868370/87778905-96ad0f00-c834-11ea-8946-54ac3393d917.png">

#### 2. 
```
psp = PerspectiveWidget(df, row_pivots=['D'], column_pivots=['A','B'],aggregates={'C':'count'},columns=['C'])
psp
```
<img width="607" alt="image" src="https://user-images.githubusercontent.com/4868370/87778999-c52aea00-c834-11ea-9086-2230769bd528.png">
So far so good...

#### 3. 
Let us change "Grid" to "Heatmap" on UI (by hand):

<img width="1382" alt="image" src="https://user-images.githubusercontent.com/4868370/87779122-f6a3b580-c834-11ea-8f46-4d816d94f2a3.png">

We see that the axis are swapped while the column names in "groupBy" and "splitBy" remained the same.
This is incorrect.

### Expected Result:
<img width="1077" alt="image" src="https://user-images.githubusercontent.com/4868370/87779318-4aae9a00-c835-11ea-9a9b-dfb992e98283.png">

### Environment:
Perspective 0.5.1

**Comment by timkpaine** [comment]
I suspect this is a duplicate of #1122, if you turn it into a flattened grid you should see the columns are all messed up to begin with.

**Comment by pgagarinov** [comment]
@timkpaine I do not think this is a duplicate of #1122 because in #1122 the source of the problem is passing pivoted dataframes into perspective. Here I do not pass pivoted dataframes, instead I use perspective for aggregation. And it works fine for "Datagrid" mode (but not for Heatmap mode).

**Comment by texodus** [comment]
@pgagarinov Thanks for the bug report!

This is expected behavior.  `group_by` is the engine's primary axis (both visually and internally) and belongs on the chart's primary axis, like the other charts.  We recently implemented #2113 for a similar issue which further diverges default visualization from grids, specifically that the default sort order for the "Y axis"  (in quotes because grids aren't exactly cartesian) is inverted for charts with the origin in the bottom left (rather than top left).

You can alternatively use the datagrid's color-by-gradient feature to get heatmap coloring on a default datagrid.

We could consider a feature request to implement "X Heatmap" variant, but this seems like overkill to me ATM as its functionally equivalent to clicking "swap". We're working through some design ideas that may also lead to deeper chart configuration in the future, and I could imagine e.g. merging the X- and Y- variant charts into a simple chart axis swap setting.
